src/pd/retry.rs

```diff
@@ -108,24 +115,81 @@ impl fmt::Debug for RetryClient {
     }
 }
 
+struct ShouldReconnect<'a> {
```

Comment:
Could you please add some comments to it?

Reply:
Please add a space after `//`.
I actually expect you can document what's the return value of this future...